plugins/woocommerce/src/Admin/API/Reports/FilteredGetDataTrait.php

```diff
@@ -0,0 +1,58 @@
+<?php
+declare( strict_types = 1);
+
+namespace Automattic\WooCommerce\Admin\API\Reports;
+
+// Exit if accessed directly.
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+
+/**
+ * Trait to call filters on `get_data` methods for data stores.
+ *
+ * It calls the filters `woocommerce_analytics_{$this->context}_query_args` and
+ * `woocommerce_analytics_{$this->context}_select_query` on the `get_data` method.
+ *
+ * Example:
+ * <pre><code class="language-php">class MyStatsDataStore extends DataStore implements DataStoreInterface {
+ *     // Use the trait.
+ *     use FilteredGetDataTrait;
+ *     // Provide all the necessary properties and methods for a regular DataStore.
+ *     // ...
+ * }
+ * </code></pre>
+ *
+ * @see DataStore
+ */
+trait FilteredGetDataTrait {
```

Comment:
I don’t see this trait being used anywhere in the PR. Is it intended for an upcoming PR, or was it used in a different PR and is showing here due to merging?

Reply:
The intent is to use in for the new classes and for 3PDs.

The existing controllers will still use `Query` classes for backward compatibility

Reply:
See the discussions at https://github.com/woocommerce/woocommerce/pull/50749 as it was reviewed there.

Sorry, for merging it before review of this one is finished :(